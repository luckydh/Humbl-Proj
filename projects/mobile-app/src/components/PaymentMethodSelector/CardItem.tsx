import React from "react";
import cx from "classnames";
import Chevron from "assets/svgs/Chevron";
import ContentLoader from "react-content-loader";
import cardPlaceholder from "assets/svgs/credit-card-placeholder.svg";

interface CardItemProps {
  last4?: string;
  brandName?: string;
  brandImage?: string;
  bordered?: boolean;
  showChevron?: boolean;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
  className?: string;
  disabled?: boolean;
}

export const CardItem: React.FC<CardItemProps> = ({
  onClick,
  last4,
  brandName,
  brandImage,
  bordered = true,
  showChevron = false,
  className,
  disabled = false,
}) => {
  const buttonClasses = cx(
    "flex mb-4 rounded-lg select-none bg-none outline-none items-center",
    bordered ? "border-white border-2 border-solid p-3 py-4" : "",
    className
  );
  return (
    <button onClick={onClick} className={buttonClasses} disabled={disabled}>
      <img src={brandImage || cardPlaceholder} alt={`Payment Brand Logo - ${brandName} `} style={{ height: 38 }} />
      <div className="ml-3 rounded text-left w-full text-white flex items-center truncate">
        <div>
          <h2 className="text-md leading-none font-bold truncate">{brandName}</h2>
          {last4 && <span className="text-base leading-none font-medium truncate">Credit - {last4}</span>}
        </div>
        {showChevron && <Chevron width={38} />}
      </div>
    </button>
  );
};

export default CardItem;

export const CardItemSkeleton: React.FC<{
  forgroundColor?: string;
  backgroundColor?: string;
}> = ({ forgroundColor, backgroundColor }) => (
    <ContentLoader
      animate={true}
      speed={2}
      width={326}
      height={38}
      viewBox="0 0 326 38"
      backgroundColor={backgroundColor || "#4cb4dd"}
      foregroundColor={forgroundColor || "#127aa3"}
      uniqueKey="card-item-skeleton">
      <rect x="0" y="0" rx="3" ry="3" width="60" height="38" />
      <rect x="75" y="0" rx="3" ry="3" width="80" height="16" />
      <rect x="75" y="22" rx="3" ry="3" width="120" height="16" />
    </ContentLoader>
  );
