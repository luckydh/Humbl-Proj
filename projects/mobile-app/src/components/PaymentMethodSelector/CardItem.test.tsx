import CardItem from "./CardItem";
import React from "react";
import renderer from "react-test-renderer";

it("renders card item without chevron", () => {
  const tree = renderer.create(<CardItem brandName="Visa" last4="1234" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders card item with chevron", () => {
  const tree = renderer
    .create(<CardItem brandName="Visa" last4="1234" showChevron={true} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders card item with custom card brand image", () => {
  const tree = renderer
    .create(
      <CardItem
        brandName="Visa"
        last4="1234"
        brandImage="https://res.cloudinary.com/humbldata/image/upload/v1615500991/payment_method_assets/VISA_kigub0.png"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
