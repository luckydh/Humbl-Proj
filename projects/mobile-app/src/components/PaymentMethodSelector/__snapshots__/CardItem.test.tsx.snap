// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders card item with chevron 1`] = `
<button
  className="flex mb-4 rounded-lg select-none bg-none outline-none items-center border-white border-2 border-solid p-3 py-4"
  disabled={false}
>
  <img
    alt="Payment Brand Logo - Visa "
    src="credit-card-placeholder.svg"
    style={
      Object {
        "height": 38,
      }
    }
  />
  <div
    className="ml-3 rounded text-left w-full text-white flex items-center truncate"
  >
    <div>
      <h2
        className="text-md leading-none font-bold truncate"
      >
        Visa
      </h2>
      <span
        className="text-base leading-none font-medium truncate"
      >
        Credit - 
        1234
      </span>
    </div>
    <svg
      className="flex-none ml-auto"
      height="100%"
      viewBox="0 0 37.996 38.002"
      width={38}
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <style>
          .a_chevron_filled{fill:#d3eff9}
        </style>
      </defs>
      <path
        className="a_chevron_filled"
        d="M19033,21472a18.944,18.944,0,1,1,7.395-1.494A18.888,18.888,0,0,1,19033,21472Zm-2.77-29.555-2.6,2.594,7.533,7.535-7.533,7.533,2.6,2.6,10.137-10.135Z"
        transform="translate(-19014.002 -21433.998)"
      />
    </svg>
  </div>
</button>
`;

exports[`renders card item with custom card brand image 1`] = `
<button
  className="flex mb-4 rounded-lg select-none bg-none outline-none items-center border-white border-2 border-solid p-3 py-4"
  disabled={false}
>
  <img
    alt="Payment Brand Logo - Visa "
    src="https://res.cloudinary.com/humbldata/image/upload/v1615500991/payment_method_assets/VISA_kigub0.png"
    style={
      Object {
        "height": 38,
      }
    }
  />
  <div
    className="ml-3 rounded text-left w-full text-white flex items-center truncate"
  >
    <div>
      <h2
        className="text-md leading-none font-bold truncate"
      >
        Visa
      </h2>
      <span
        className="text-base leading-none font-medium truncate"
      >
        Credit - 
        1234
      </span>
    </div>
  </div>
</button>
`;

exports[`renders card item without chevron 1`] = `
<button
  className="flex mb-4 rounded-lg select-none bg-none outline-none items-center border-white border-2 border-solid p-3 py-4"
  disabled={false}
>
  <img
    alt="Payment Brand Logo - Visa "
    src="credit-card-placeholder.svg"
    style={
      Object {
        "height": 38,
      }
    }
  />
  <div
    className="ml-3 rounded text-left w-full text-white flex items-center truncate"
  >
    <div>
      <h2
        className="text-md leading-none font-bold truncate"
      >
        Visa
      </h2>
      <span
        className="text-base leading-none font-medium truncate"
      >
        Credit - 
        1234
      </span>
    </div>
  </div>
</button>
`;
